azure-pipelines/build.yml

```diff
@@ -49,6 +49,22 @@ steps:
     codeCoverageTool: 'Cobertura'
     summaryFileLocation: '$(Agent.TempDirectory)/coverlet/reports/Cobertura.xml'
 
+- task: PowerShell@2
+  displayName: 'Generate Nuspec Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $propsFile = '$(Build.SourcesDirectory)\Directory.Build.Props'
+      $props = New-Object XML
+      $props.Load($propsFile)
+      $propGroup = $props.Project.PropertyGroup;
+      $version = $propGroup.VersionPrefix + '-' + 'preview' + $(Build.BuildNumber)
```

Comment:
As David noted above, the preview should generally be the last part of the version string. So `3.0.0.1234-preview` instead of `3.0.0-preview1234`